src/plugins/intel_gpu/src/kernel_selector/kernels/grid_sample/grid_sample_kernel_opt_bilinear_zeros.cpp

```diff
@@ -31,13 +31,25 @@ bool GridSampleKernelOpt_BilinearZeros::Validate(const Params& params) const {
     if (!TBase::Validate(params))
         return false;
 
+    auto PaddedSpatial = [](const MultiDataTensor& tensors) -> bool {
+        bool is_padded = false;
+        for (auto tensor : tensors) {
+            is_padded |= tensor.X().pad.Total() != 0;
+            is_padded |= tensor.Y().pad.Total() != 0;
```

Comment:
this code assumes 4d data only. could you add assertion for that it is actually 4d? (I mean, it does not have handling for `Z`)

Reply:
Added assertion for checking 4dims.